File: C#/Climbing Stairs.cs
public class Solution70 {
    public int ClimbStairs(int n) {
        if(n<=1) return 1;
        
        int pre = 1;
        int curr = 1;
        
        for(int i=2;i<=n;i++)
        {
            int temp = curr;
            curr = pre + curr;
            pre = temp;
        }
        
        return curr;
    }
}

File: Golang/Climbing Stairs.go
func climbStairs(n int) int {
    n1 := 2
    n2 := 3
    if n <= 3 {
        return n
    }

    for i:=4;i < n + 1;i++ {
        temp := n1 + n2
        n1 = n2
        n2 = temp
    }
    return n2
}